### DISCUSSION TITLE
Build on Mac

### DISCUSSION THREAD

**Discussion by codebravely** [snesrev/zelda3#111] [Q&A] [answered]
Hello there,

I was very interested in building this on a Mac, and well done on such a great project, but it seems I've hit a stumbling block while trying to compile.  

Just after running 'make', I get the following error:

`[%]+make`
`cc -c -O2 -I/opt/homebrew/include/SDL2 -D_THREAD_SAFE hud.c -o hud.o`
`hud.c:17:13: error: static declaration of 'Hud_DoWeHaveThisItem' follows non-static declaration`
`static bool Hud_DoWeHaveThisItem(uint8 item);`
`            ^`
`./hud.h:31:6: note: previous declaration is here`
`bool Hud_DoWeHaveThisItem(uint8 item);`
`     ^`
`hud.c:524:13: error: static declaration of 'Hud_DoWeHaveThisItem' follows non-static declaration`
`static bool Hud_DoWeHaveThisItem(uint8 item) {  // 8ddeb0`
 `           ^`
`./hud.h:31:6: note: previous declaration is here`
`bool Hud_DoWeHaveThisItem(uint8 item);`
`     ^`
`2 errors generated.`
`make: *** [hud.o] Error 1`

Thanks very much.

**Comment by UltraHDR** [comment]
Works fine with macOS 12.6 M1 Pro. What macOS version? What CPU? Did you run the Python scripts first? Also did you install brew?

Please run `make clean` and post all your compile log

**Reply by codebravely** [reply]
macOS version is 12.6 and I'm on an original M1 macbook.

Yes, I ran the Python scripts first and have had brew installed for years. I'll try to ensure all the dependencies are correct again.

I will try a `make clean`

Thanks for the help.

**Reply by codebravely** [reply]
I used the latest master and ran make clean, then make and all is working correctly.  Thanks for the help!

**Comment by snesrev** [comment, marked as answer]
Please make sure you use the latest master. I fixed something the other day.

**Reply by codebravely** [reply]
Will check and try again, I did just try to build this from master 2 days ago. Thank you.

**Reply by codebravely** [reply]
Must have been an old version, the new master worked after an update, make clean and make.

Thank you.
